fastapi-challenges/file-upload-api/devashmit-day28/main.py

```
@@ -0,0 +1,33 @@
+class UpRes(BaseModel): original_name:str; thumbnail_url:str; medium_url:str; large_url:str
+
+@app.post("/upload", response_model=UpRes, status_code=201)
+async def upload(file: UploadFile):
+    if file.content_type not in ALLOWED: raise HTTPException(415, "JPEG/PNG/WebP only")
+    data = await file.read()
+    if len(data) > MAX: raise HTTPException(413, "Max 5MB")
+    folder = ROOT / str(date.today()); folder.mkdir(parents=True, exist_ok=True)
+    base = secrets.token_hex(8)
+    try: img = Image.open(io.BytesIO(data))
+    except Exception as e: raise HTTPException(400, f"Invalid image: {e}")
+    if img.mode in ("RGBA","P"): img = img.convert("RGB")
```

Ruff (0.15.18)

[error] Multiple statements on one line (colon)

(E701)